src/main/scala/io/github/spark_redshift_community/spark/redshift/v2/RedshiftPreProcessor.scala

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.spark_redshift_community.spark.redshift.v2
+
+import java.io.InputStreamReader
+import java.net.URI
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+import com.amazonaws.auth.AWSCredentialsProvider
+import com.amazonaws.services.s3.AmazonS3Client
+import com.eclipsesource.json.Json
+import io.github.spark_redshift_community.spark.redshift.{AWSCredentialsUtils, DefaultJDBCWrapper, FilterPushdown, JDBCWrapper, Utils}
+import io.github.spark_redshift_community.spark.redshift.Parameters.MergedParameters
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.sources.Filter
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{SaveMode, SparkSession}
+import scala.collection.JavaConverters._
+
+import com.amazonaws.services.s3.model.{AmazonS3Exception, ListObjectsV2Request}
+
+class RedshiftPreProcessor(schemaOpt: Option[StructType],
+    requiredSchema: StructType,
+    params: MergedParameters,
+    pushedFilters: Array[Filter]) extends Logging {
+
+  val jdbcWrapper: JDBCWrapper = DefaultJDBCWrapper
+
+  private def buildUnloadStmt(
```

Comment:
related to the other comment about v1 sources supporting ttl and parquet.  iirc there is a build unload stmt method somewhere already and we might be able to share code between the v1 and v2 sources to have a single unload stmt builder?

Reply:
Yeah refactoring this would be valuable